crates/net/network/src/fetch/mod.rs

```diff
@@ -103,11 +114,17 @@ impl StateFetcher {
         }
     }
 
-    /// Returns the _next_ idle peer that's ready to accept a request.
+    /// Returns the _next_ idle peer that's ready to accept a request,
+    /// prioritizing those with the lowest timeout/latency.
     /// Once a peer has been yielded, it will be moved to the end of the map
     fn next_peer(&mut self) -> Option<PeerId> {
-        let peer =
-            self.peers.iter().find_map(|(peer_id, peer)| peer.state.is_idle().then_some(*peer_id));
+        let peer = self
+            .peers
+            .iter()
+            .filter(|(_, peer)| peer.state.is_idle())
+            .min_by_key(|(_, peer)| peer.timeout.load(Ordering::Relaxed).to_owned())
+            .map(|(id, _)| id.to_owned());
+
         if let Some(peer_id) = peer {
             // Move to end of the map
             self.peers.get_refresh(&peer_id);
```

Comment:
this is now redundant, right?

so perhaps we can change this to a `HashMap` again

Reply:
I left it so we alternate peers with equal timeout ([see test_peer_rotation](https://github.com/paradigmxyz/reth/blob/67402ed45b6474e570cee87ddf3985a1a0c332d5/crates/net/network/src/fetch/mod.rs#L414)), but maybe it's not useful enough to keep. wdyt?

Reply:
yeh I think no longer useful if we're now using the timeout, unlikely that multiple peers will have the same timeout, right?

Reply:
Totally. I'll make this change then.